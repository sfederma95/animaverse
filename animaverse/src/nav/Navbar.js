import React from 'react';
import {NavLink} from 'react-router-dom'

function Navbar(){
    return(
        <ul>
            <li>
                <NavLink to='/login'>
                    Login
                </NavLink>
            </li>
            <li>
                <NavLink to='/register'>
                    Sign Up
                </NavLink>
            </li>
            <li>
                <NavLink to='/questions'>
                    FAQ
                </NavLink>
            </li>
        </ul>
    )
}

export default Navbar;