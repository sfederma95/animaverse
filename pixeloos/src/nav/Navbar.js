import React, {useContext} from 'react';
import {NavLink} from 'react-router-dom';
import UserContext from '../users/UserContext';
import './nav.css'
import homeBtn from './home-btn.svg'
import {disappear, homeBtnBack} from './nav'

function Navbar({logout}){
    const {currentUser} = useContext(UserContext)
    function logoutBtn(){
        homeBtnBack();
        logout();
    }
    function loggedIn(){
        return(
                <div className='links-div'>
                    <NavLink onClick={homeBtnBack} className='nav-li' to={`/users/${currentUser.usr_id}`}>
                        Profile
                    </NavLink>
                    <NavLink onClick={homeBtnBack} className='nav-li' to={`/${currentUser.usr_id}/game`}>
                        Play Game
                    </NavLink>

                    <NavLink onClick={logoutBtn} className='nav-li' to='/'>
                        Logout
                    </NavLink>
                </div>
        )
    }
    function loggedOut(){
        return(
            <div className='links-div'>
                    <NavLink onClick={homeBtnBack} className='nav-li' to='/'>
                        Home
                    </NavLink>
                    <NavLink onClick={homeBtnBack} className='nav-li' to='/login'>
                        Login
                    </NavLink>
                    <NavLink onClick={homeBtnBack} className='nav-li' to='/register'>
                        Sign Up
                    </NavLink>
            </div>
            
        )
    }

    return(
        <nav id='navbar'>
            {currentUser ? loggedIn() : loggedOut()}
            <img onClick={disappear} alt='home button' id='home-btn' src={homeBtn}/>
        </nav>
    )
}

export default Navbar;