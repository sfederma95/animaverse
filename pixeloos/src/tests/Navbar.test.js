import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import {MemoryRouter} from 'react-router';
import Navbar from '../nav/Navbar';
import {UserProvider} from './testSetup';
// import $ from 'jquery';
// window.jQuery = $;
// window.$ = $;

it ('renders nav component without crashing', function(){
    render(
        <MemoryRouter>
            <UserProvider>
                <Navbar />
            </UserProvider>
        </MemoryRouter>
    )
})

it ('matches snapshot when logged in', function(){
    const {asFragment} = render (
        <MemoryRouter>
            <UserProvider>
                <Navbar/>
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})

it ('matches snapshot when logged out', function(){
    const {asFragment} = render(
        <MemoryRouter>
            <UserProvider currentUser={null}>
                <Navbar />
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})

it ('hides home button appropriately', function(){
    const {container} = render(
        <MemoryRouter>
            <UserProvider>
                <Navbar/>
            </UserProvider>
        </MemoryRouter>
    );
    expect(container.getElementsByClassName('nav-li')[0]).toHaveStyle('opacity:0%');
    const hmbtn = container.querySelector('#home-btn');
    fireEvent.click(hmbtn);
    expect(hmbtn).not.toBeVisible();
})