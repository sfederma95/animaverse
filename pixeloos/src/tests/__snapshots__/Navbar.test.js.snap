// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when logged in 1`] = `
<DocumentFragment>
  <nav
    id="navbar"
  >
    <div
      class="links-div"
    >
      <a
        class="nav-li"
        href="/users/7"
      >
        Profile
      </a>
      <a
        class="nav-li"
        href="/7/game"
      >
        Play Game
      </a>
      <a
        aria-current="page"
        class="nav-li active"
        href="/"
      >
        Logout
      </a>
    </div>
    <img
      alt="home button"
      id="home-btn"
      src="home-btn.svg"
    />
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <nav
    id="navbar"
  >
    <div
      class="links-div"
    >
      <a
        aria-current="page"
        class="nav-li active"
        href="/"
      >
        Home
      </a>
      <a
        class="nav-li"
        href="/login"
      >
        Login
      </a>
      <a
        class="nav-li"
        href="/register"
      >
        Sign Up
      </a>
    </div>
    <img
      alt="home button"
      id="home-btn"
      src="home-btn.svg"
    />
  </nav>
</DocumentFragment>
`;
